pymobiledevice3/__main__.py

```diff
@@ -64,62 +31,58 @@
-    except NotPairedError:
-        logger.error('Device is not paired')
-    except UserDeniedPairingError:
-        logger.error('User refused to trust this computer')
-    except PairingDialogResponsePendingError:
-        logger.error('Waiting for user dialog approval')
-    except SetProhibitedError:
-        logger.error('lockdownd denied the access')
-    except MissingValueError:
-        logger.error('No such value')
-    except DeviceHasPasscodeSetError:
-        logger.error('Cannot enable developer-mode when passcode is set')
-    except DeveloperModeError as e:
-        logger.error(f'Failed to enable developer-mode. Error: {e}')
-    except ConnectionFailedToUsbmuxdError:
-        logger.error('Failed to connect to usbmuxd socket. Make sure it\'s running.')
-    except MessageNotSupportedError:
-        logger.error('Message not supported for this iOS version')
-        traceback.print_exc()
-    except InternalError:
-        logger.error('Internal Error')
-    except DeveloperModeIsNotEnabledError:
-        logger.error('Developer Mode is disabled. You can try to enable it using: '
-                     'python3 -m pymobiledevice3 amfi enable-developer-mode')
-    except InvalidServiceError:
-        logger.error(INVALID_SERVICE_MESSAGE)
-    except NoDeviceSelectedError:
-        return
-    except PasswordRequiredError:
-        logger.error('Device is password protected. Please unlock and retry')
-    except AccessDeniedError:
-        logger.error('This command requires root privileges. Consider retrying with "sudo".')
-    except BrokenPipeError:
-        traceback.print_exc()
-    except TunneldConnectionError:
-        logger.error(
-            'Unable to connect to Tunneld. You can start one using:\n'
-            'sudo python3 -m pymobiledevice3 remote tunneld')
-    except DeviceNotFoundError as e:
-        logger.error(f'Device not found: {e.udid}')
+# Mapping of index options to import file names
+CLI_GROUPS = {
+    "activation": "activation",
+    "afc": "afc",
+    "amfi": "amfi",
+    "apps": "apps",
+    "backup2": "backup",
+    "bonjour": "bonjour",
+    "companion": "companion_proxy",
+    "crash": "crash",
+    "developer": "developer",
+    "diagnostics": "diagnostics",
+    "lockdown": "lockdown",
+    "mounter": "mounter",
+    "notification": "notification",
+    "pcap": "pcap",
+    "power-assertion": "power_assertion",
+    "processes": "processes",
+    "profile": "profile",
+    "provision": "provision",
+    "remote": "remote",
+    "restore": "restore",
+    "springboard": "springboard",
+    "status": "status",
+    "syslog": "syslog",
+    "usbmux": "usbmux",
+    "webinspector": "webinspector"
+}
 
 
+class Pmd3Cli(click.Group):
+    def list_commands(self, ctx):
+        return CLI_GROUPS.keys()
+
+    def get_command(self, ctx, name):
+        try:
+            mod = __import__(f"pymobiledevice3.cli.{CLI_GROUPS[name]}", None, None, ['cli'])
+        except ImportError:
```

Comment:
In order to preserve old behavior please raise the proper click error.
For example when I run:

```shell
pymobiledevice3 non-existing-subcommand
```

I'm expecting:
```
Usage: pymobiledevice3 [OPTIONS] COMMAND [ARGS]...
Try 'pymobiledevice3 -h' for help.

Error: No such command 'non-existing-subcommand'.
```

Reply:
Fixed with [8dcbc58](https://github.com/doronz88/pymobiledevice3/pull/702/commits/8dcbc5840832e170fa062fb119d514c4fd32e57b)